packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -329,17 +329,26 @@ const getGraphqlData = async (ipfsHash: string, url: string, retryOptions: Retry
 
 // We don't want to throw an error if the IPFS request fails for any reason, so we return an Error object instead that
 // will be logged by the bot.
-const getIpfsData = async (ipfsHash: string, url: string, retryOptions: RetryOptions): Promise<unknown | Error> => {
+const getIpfsData = async (
+  ipfsHash: string,
+  url: string,
+  strictValidation: boolean,
+  retryOptions: RetryOptions
+): Promise<unknown | Error> => {
   let response: Response;
 
   // Separate try/catch block to catch errors from the fetch() call itself.
   try {
     response = await retry(async () => await fetch(`${url}/${ipfsHash}`), retryOptions);
   } catch (error) {
     assert(error instanceof Error, "Unexpected Error type!");
-    return new IpfsFetchError(error);
+    return new IpfsFetchError(error); // This happens when the IPFS gateway is not returning HTTP response.
+  }
+  if (!response.ok) {
+    const validationError = new Error(`Request on ${response.url} failed with status ${response.status}`);
+    if (strictValidation) return validationError;
+    return new IpfsFetchError(validationError); // Caller would detect the wrapped error to identify it as server issue.
```

Comment:
So the case is IPFS response is valid, but it's an error response. The case in the above catch is where the response is completely invalid and can't be parsed.

Is that right?

Are there any error codes or other sub-types we can trigger on? (We can do this in a follow-up)

Specifically, I'm thinking of trying to differentiate things like not finding the IPFS document (maybe a 4xx) vs a server error (5xx) or similar.